src/symbol/annotations.rs

```diff
@@ -0,0 +1,290 @@
+use std::result;
+
+use crate::common::*;
+use crate::FallibleIterator;
+
+/// These values correspond to the BinaryAnnotationOpcode enum from the
+/// cvinfo.h
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+enum BinaryAnnotationOpcode {
+    /// Link time pdb contains PADDINGs.
+    ///
+    /// These are represented with the 0 opcode which is in some PDB
+    /// implementation called "invalid".
+    Eof = 0,
+    /// param : start offset
+    CodeOffset = 1,
+    /// param : nth separated code chunk (main code chunk == 0)
+    ChangeCodeOffsetBase = 2,
+    /// param : delta of offset
+    ChangeCodeOffset = 3,
+    /// param : length of code, default next start
+    ChangeCodeLength = 4,
+    /// param : fileId
+    ChangeFile = 5,
+    /// param : line offset (signed)
+    ChangeLineOffset = 6,
+    /// param : how many lines, default 1
+    ChangeLineEndDelta = 7,
+    /// param : either 1 (default, for statement)
+    ///         or 0 (for expression)
+    ChangeRangeKind = 8,
+    /// param : start column number, 0 means no column info
+    ChangeColumnStart = 9,
+    /// param : end column number delta (signed)
+    ChangeColumnEndDelta = 10,
+    /// param : ((sourceDelta << 4) | CodeDelta)
+    ChangeCodeOffsetAndLineOffset = 11,
+    /// param : codeLength, codeOffset
+    ChangeCodeLengthAndCodeOffset = 12,
+    /// param : end column number
+    ChangeColumnEnd = 13,
+    /// A non valid value
+    Invalid,
+}
+
+impl From<u32> for BinaryAnnotationOpcode {
+    fn from(value: u32) -> Self {
+        match value {
+            0 => BinaryAnnotationOpcode::Eof,
+            1 => BinaryAnnotationOpcode::CodeOffset,
+            2 => BinaryAnnotationOpcode::ChangeCodeOffsetBase,
+            3 => BinaryAnnotationOpcode::ChangeCodeOffset,
+            4 => BinaryAnnotationOpcode::ChangeCodeLength,
+            5 => BinaryAnnotationOpcode::ChangeFile,
+            6 => BinaryAnnotationOpcode::ChangeLineOffset,
+            7 => BinaryAnnotationOpcode::ChangeLineEndDelta,
+            8 => BinaryAnnotationOpcode::ChangeRangeKind,
+            9 => BinaryAnnotationOpcode::ChangeColumnStart,
+            10 => BinaryAnnotationOpcode::ChangeColumnEndDelta,
+            11 => BinaryAnnotationOpcode::ChangeCodeOffsetAndLineOffset,
+            12 => BinaryAnnotationOpcode::ChangeCodeLengthAndCodeOffset,
+            13 => BinaryAnnotationOpcode::ChangeColumnEnd,
+            _ => BinaryAnnotationOpcode::Invalid,
```

Comment:
Would it be better to use `TryFrom` here? That would let us drop the `Invalid` enum member, making every `BinaryAnnotationOpcode` a recognized opcode again.

Reply:
https://github.com/willglynn/pdb/pull/47#discussion_r320676309